src/lockscreen.js

```diff
@@ -201,8 +201,19 @@ var Lockscreen = GObject.registerClass({
 
         this._updateBackground();
 
-        if (!this._manager.isUnlocked(this._lock))
+        try {
+            if (!this._manager.isUnlocked(this._lock))
+                return;
+        } catch (e) {
+            logError(e, 'Error trying to load unlock state, trying again in 100 milisecond');
+            GLib.timeout_add(GLib.PRIORITY_HIGH, 100,
+                () => {
+                    this._updateLockStateWithLock();
+                    return GLib.SOURCE_REMOVE;
+                });
```

Comment:
This means this function will be tried until it succeeds. Should we limit the number of attempts? What happens if it's still trying and the user quits the application/toolbox?

Reply:
Yes, you're right, if the service is broken or not installed for some reason this will be trying until the end of the app. I'll add a number of tries, like 3 for example.